spec/template.dd

```diff
@@ -1135,23 +1140,38 @@ $(H2 $(LNAME2 function-templates, Function Templates))
         foo(cptr);  // T == const(int)*
         foo(iptr);  // T == immutable(int)*
         ------
-
-    $(P Type parameter deduction is not influenced by the order of function
-        arguments.
     )
 
+$(H3 $(LNAME2 return-deduction, Return Type Deduction))
+
     $(P Function templates can have their return types deduced based on the
         $(GLINK2 statement, ReturnStatement)s in the function, just as with
         normal functions.
         See $(DDSUBLINK function, auto-functions, Auto Functions).
     )
 
+    $(SPEC_RUNNABLE_EXAMPLE_COMPILE
         ---
-        auto $(CODE_HIGHLIGHT Square)(T)(T t)
+        auto $(CODE_HIGHLIGHT square)(T)(T t)
```

Comment:
Test failed.
```
__stdin.d(3): Error: basic type expected, not `$`
__stdin.d(3): Error: found `$` instead of statement
__stdin.d(11): Error: unrecognized declaration
```
Shouldn't `CODE_HIGHLIGHT` be removed here?

Reply:
Yes it should be. Ping @ntrel . Removed `auto-merge` for the time being, ping me when the CI is fixed.

Reply:
@Geod24 Now fixed. I ran the example in the browser, which worked, but I hadn't seen it broke `test_dspec`. circleci is now failing on `web/phobos/dmd_arraytypes.html`, but I think that's unrelated.